File: samplecode/go/sample.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"crypto/tls"
	"encoding/hex"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"net/url"
	"sort"
	"strings"
	"time"
)

func calculateAuthorizationHeader(requestURL, requestTimeStamp, keyID, keyValue, httpMethod string) string {
	u, err := url.Parse(requestURL)
	if err != nil {
		panic(err)
	}

	var path = u.Path
	m, _ := url.ParseQuery(u.RawQuery)

	var keys []string
	for k := range m {
		keys = append(keys, k)
	}
	sort.Strings(keys)
	var orderedQueries []string
	for _, k := range keys {
		orderedQueries = append(orderedQueries, fmt.Sprintf("%s:%s", k, m[k][0]))
	}

	var queries = strings.Join(orderedQueries, ", ")
	content := fmt.Sprintf("%s\r\n%s\r\n%s\r\n%s", path, queries, requestTimeStamp, httpMethod)
	fmt.Print(content)
	hash := hmac.New(sha256.New, []byte(keyValue))
	hash.Write([]byte(content))
	digest := strings.ToUpper(hex.EncodeToString(hash.Sum(nil)))
	return fmt.Sprintf("AzureCDN %s:%s", keyID, digest)
}

func main() {
	apiURL := "-- your request api url --"
	keyID := "-- your request api key id --"
	keyValue := "-- your request api key value --"
	requestMethod := strings.ToUpper("-- your request method --")
	requestTimestamp := time.Now().UTC().Format(time.RFC3339)

	client := &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				Renegotiation: tls.RenegotiateOnceAsClient,
				Certificates:  nil,
			},
		},
	}

	req, ex := http.NewRequest(requestMethod, apiURL, nil)
	if ex != nil {
		log.Fatal(ex)
	}

	// Add headers
	req.Header.Set("content-type", "application/json")
	req.Header.Set("Authorization", calculateAuthorizationHeader(apiURL, requestTimestamp, keyID, keyValue, "GET"))
	req.Header.Set("x-azurecdn-request-date", requestTimestamp)

	resp, ex := client.Do(req)
	if ex != nil {
		log.Fatal(ex)
	}

	body, _ := ioutil.ReadAll(resp.Body)
	response := string(body)
	fmt.Print(response)
}

File: samplecode/python/sample.py
import requests, json
import urllib
import datetime
import hmac, hashlib, base64
from collections import OrderedDict

def pp_json(json_thing, sort=True, indents=4):
    if type(json_thing) is str:
        print(json.dumps(json.loads(json_thing), sort_keys=sort, indent=indents))
    else:
        print(json.dumps(json_thing, sort_keys=sort, indent=indents))
    return None

def calculate_authorization_header(request_url, request_timestamp, key_id, key_value, http_method):
    urlparts = urllib.parse.urlparse(request_url)
    queries = urllib.parse.parse_qs(urlparts.query)
    ordered_queries = OrderedDict(queries)

    message = "%s\r\n%s\r\n%s\r\n%s" % (urlparts.path, ", ".join(['%s:%s' % (key, value[0]) for (key, value) in ordered_queries.items()]), request_timestamp, http_method)
    digest = hmac.new(bytearray(key_value, "utf-8"), bytearray(message, "utf-8"), hashlib.sha256).hexdigest().upper()
    return "AzureCDN %s:%s" % (key_id, digest)

apiUrl = "-- your requst url --"
request_timestamp = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
key_id = "-- your request api key id --"
key_value = "-- your request api key value --"
method = "--your request method--".upper()

headers = {
             'content-type': 'application/json',
             'Authorization': calculate_authorization_header(apiUrl, request_timestamp, key_id, key_value, "GET"),
             'x-azurecdn-request-date': request_timestamp
          }
          
r = requests.get(apiUrl, headers = headers) 
print(r.headers)
print(pp_json(r.text))